fabric/fabric_ddp.py

```diff
@@ -1,3 +1,48 @@
 import lightning as L
+import torch
+import torch.nn.functional as F
+from lightning.pytorch.demos import Transformer, WikiText2
+from torch.utils.data import DataLoader, random_split
 
-print('hello lightning')
\ No newline at end of file
+
+def main():
+    L.seed_everything(42)
+
+    fabric = L.Fabric(strategy="ddp")
+    fabric.launch()
+
+    # Data
+    with fabric.rank_zero_first(local=True):
+        WikiText2(download=True)
+
+    dataset = WikiText2(download=False)
```

Comment:
You could directly do:

```py
    with fabric.rank_zero_first(local=True):
        dataset = WikiText2(download=True)
```

All that matters is that the process 0 does it first, and then the others come and verify that the data is already downloaded. This is what the context manager guarantees.